### PR TITLE
PRO-2195 Outdated WalletConnect on plr.pillar.fi

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-2195/outdated-walletconnect-on-plrpillarfi

* Updated deprecated packages
* Fixed small issues
* Added web3modal
* Fixed typescript error
* Fixed lint error

https://github.com/pillarwallet/synthetix-mintr/assets/82652040/ce1e1689-08ec-4db9-be9f-8b9cdaae1931

### CODE REVIEW

**Review by bhumitaylor14 — APPROVED** [review]
LGTM!

File: src/components/Typography/DynamicComponent.js [hunk 1]
```diff
@@ -15,8 +15,8 @@ import {
 } from 'styled-system';
 
 export default styled.div`
-  ${space}
-  ${fontSize}
+	${space}
+	${fontSize}
```
Comment (IAmKio) [inline src/components/Typography/DynamicComponent.js:19]: Minor: indentation is off here

**Review by IAmKio — APPROVED** [review]
LGTM

File: src/components/Typography/DynamicComponent.js (hunk 1 above)
Comment (lalitkumarrock) [reply src/components/Typography/DynamicComponent.js:19]: without this comes prettier lint error

**Pull request merged**